great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -105,8 +179,12 @@ def get_unmatched_data_references(self) -> List[str]:
         return []
 
     def get_available_data_asset_names(self) -> List[str]:
-        """Please see note in : _get_batch_definition_list_from_batch_request()"""
-        return list(self._data_references_cache.keys())
+        """Returns a list of data_assets that are both defined at runtime, and defined in DataConnector configuration"""
+        defined_assets: List[str] = list(self.assets.keys())
+        data_reference_keys: List[str] = list(self._data_references_cache.keys())
+        available_assets: List[str] = list(set(defined_assets + data_reference_keys))
+        available_assets.sort()
```

Comment:
@Shinnnyshinshin Just a thought: this "in-place" sort is less desirable than the good old plain `Python` `sorted`.  Thanks.